File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/mvp/di/modules/MainModule.java
package com.woowrale.mvp.di.modules;

import android.content.Context;
import com.woowrale.mvp.MainApplication;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class MainModule {

    private final MainApplication mainApplication;

    public MainModule(MainApplication mainApplication) {
        this.mainApplication = mainApplication;
    }

    @Provides
    @Singleton
    Context provideContext(){
        return mainApplication;
    }

}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/mvp/di/modules/MainModule.kt
package com.woowrale.mvp.di.modules

import android.content.Context
import com.woowrale.mvp.MainApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class MainModule(val mainApplication: MainApplication) {

    init {
        val mainApplication: MainApplication
    }

    @Provides
    @Singleton
    fun provideContext(): Context {
        return mainApplication
    }
}
